File: arrays&hashing/groupAnagrams.js

var groupAnagrams = function(strs) {

    let anagrams = new Map()
    
    for (let i = 0; i < strs.length; i++) {
        let key = Array.from(strs[i]).sort().join('')
        if (anagrams.has(key)) {
            let newVal = anagrams.get(key).push(strs[i])
            continue
        }
        anagrams.set(key, [strs[i]])
    }
    return Array.from(anagrams.values());
};

File: arrays&hashing/groupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> anagrams = new HashMap<>();
        
        for (String str : strs) {
            char[] strChars = str.toCharArray();
            Arrays.sort(strChars);
            String sortedStr = new String(strChars);
            
            if (!anagrams.containsKey(sortedStr)) {
                anagrams.put(sortedStr, new ArrayList<>());
            }
            anagrams.get(sortedStr).add(str);
        }

        return new ArrayList<>(anagrams.values());

    } 
}
